cardano-cli/src/Cardano/CLI/EraBased/Transaction/Option.hs

```diff
@@ -391,17 +392,116 @@ pTransactionCalculatePlutusScriptCost :: EnvCli -> Parser (TransactionCmds era)
-      <$> ( LocalNodeConnectInfo
-              <$> pConsensusModeParams
-              <*> pNetworkId envCli
-              <*> pSocketPath envCli
-          )
+      <$> pNodeContext envCli
       <*> pTxInputFile
       <*> optional pOutputFile
  where
   pTxInputFile :: Parser FilePath
   pTxInputFile = parseFilePath "tx-file" "Filepath of the transaction whose Plutus scripts to calculate the cost."
 
+pNodeContext :: EnvCli -> Parser NodeContextInfoSource
+pNodeContext envCli = pNodeConnectionInfo <|> pLocalContext envCli
+
+pNodeConnectionInfo :: Parser NodeContextInfoSource
+pNodeConnectionInfo =
+  ProvidedTransactionContextInfo
+    <$> ( TransactionContext
+            <$> pSystemStart
+            <*> pMustExtendEraHistorySafeZone
+            <*> pEraHistoryFile
+            <*> pUtxoFile
+            <*> pProtocolParamsFile
+        )
+
+pMustExtendEraHistorySafeZone :: Parser MustExtendSafeZone
+pMustExtendEraHistorySafeZone =
+  Opt.flag'
+    MustExtendSafeZone
+    ( mconcat
+        [ Opt.long "unsafe-extend-safe-zone"
+        , Opt.help $
+            mconcat
+              [ "Allow overriding the validity of the era history past the safe zone. The "
+              , "safe zone is a period of time during which we are sure there won't be any "
+              , "era transition (hard fork), and we are confident that the slot duration "
+              , "will not change, thus the conversion from slot numbers to POSIX times "
+              , "using the era history will be correct. "
+              , "This safe zone is conservative. Even if we are past the safe zone, if "
+              , "there hasn't been any era transition (hard fork) since we obtained it, we can "
+              , "continue safely using the era history. "
+              , "This flag essentially disables the safe zone check. This allows the user to "
+              , "use the era history past the safe zone, at the user's discretion."
+              ]
+        ]
+    )
+    <|> pure DoNotExtendSafeZone
+
+pSystemStart :: Parser SystemStartOrGenesisFileSource
+pSystemStart =
+  asum
+    [ SystemStartLiteral <$> (systemStartUTC <|> systemStartPOSIX)
+    , SystemStartFromGenesisFile . GenesisFile
+        <$> parseFilePath
+          "genesis-file"
+          "Path to the Byron genesis file from which to get the start time."
+    ]
+
+systemStartPOSIX :: Parser SystemStart
+systemStartPOSIX =
+  SystemStart . posixSecondsToUTCTime . fromInteger
+    <$> ( Opt.option integralReader $
+            mconcat
+              [ Opt.long "start-time-posix"
+              , Opt.metavar "POSIX-TIME"
+              , Opt.help
+                  "The genesis start time as POSIX seconds."
+              ]
+        )
+
+systemStartUTC :: Parser SystemStart
+systemStartUTC =
+  SystemStart . convertTime
+    <$> ( Opt.strOption $
+            mconcat
+              [ Opt.long "start-time-utc"
+              , Opt.metavar "UTC-TIME"
```

Comment:
Optimally, we should use underscore instead of hyphen for metavar.

Reply:
Addressed here: https://github.com/IntersectMBO/cardano-cli/commit/914fa6b0d93514ff2f51e9e3c272abf3b57936ac